### ISSUE TITLE
Set Laravel locale based on user preference using Middleware

### ISSUE THREAD

**Issue by ArabAgile** [laravel/passport#364]
I'm not sure if middleware is the right place to do this, but found the solution here https://laracasts.com/discuss/channels/general-discussion/where-to-setlocale-in-laravel-5-on-multilingual-multidomain-app to inject the `setLocale` method. But user always returned as `null`!

The steps I did is:

I created the following `UserLocale` middleware, and added it at the end of ` $middleware` array in `Kernel.php` file:

```
<?php

namespace App\Http\Middleware;

use Closure;
use Illuminate\Support\Facades\App;
use Illuminate\Support\Facades\Auth;
use Illuminate\Support\Facades\Session;

/**
 * Class UserLocale
 * @package App\Http\Middleware
 * Short description for class
 *
 * Locale Middleware
 *
 * @since      Class available since Release 1.0.0
 */
class UserLocale
{

    /**
     * Allowed languages
     * @var array
     */
    protected $languages = ['en', 'ar'];


    /**
     * Handle an incoming request.
     *
     * @param  \Illuminate\Http\Request  $request
     * @param  \Closure  $next
     * @return mixed
     */
    public function handle($request, Closure $next)
    {
        
        dd($request->user()); // This is null!
        
        if($user = Auth::user())
        {
            if (!empty($user->language) && in_array($user->language, $this->languages)) {
                $locale = $user->language;
                App::setLocale($locale);
            }
        }

        return $next($request);
    }
}
```

And the route definition is:
```
Route::group(['middleware' => 'auth:api'], function () {

    Route::get('test', function() {
        return [
            'user' => Auth::user(),
            'locale' => App::getLocale(),
        ];
    });

});
```

now when calling `GET /test`, and pass the correct token, it shows the DEFAULT locale in `app.locale` !

My question is: why `dd($request->user());` and `if($user = Auth::user())` are always returns `null`? I can't detect the user is logged in from the middleware? and wonder how `Laravel\Passport\Http\Middleware\CheckScopes` is working!

I need to detect user language EARLY, before reaching the controller.

Again, in `UsersController`, the `Auth::user()` is `null` in the constructor `__construct()`, but I could get the logged in user from within any method like `UsersController@showShowthing`.

Any advice?!

**Comment by ArabAgile** [comment]
I found a solution for this.

I created new Middleware `UserLocale` and in kerne's `$middlewareGroups` defined it as following:

```
protected $middlewareGroups = [
        'web' => [
           ...
        ],
        'my-api' => [
            'locale',
        ],
    ];
```

And defined `Locale` in `$routeMiddleware` as:

```
protected $routeMiddleware = [
       ...
        // Show language as preferred by user
        'locale' => UserLocale::class,

    ];
```

In this case, the middleware for each api call will be `my-api`

**Comment by abdullahbutt** [comment]
Amazing & complete solution available on link below:

https://medium.com/teknomuslim/multi-language-api-response-messages-in-laravel-5c9029a32e5c

I have just implemented in Laravel 5.5. Amazing Article
